src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs

```diff
@@ -91,6 +91,20 @@ public static string GetFullPath(EnvDTE.Project envDTEProject)
             // for start up scenarios such as VS Templates. In these cases we need to fallback 
             // until we can find one containing the full path.
 
+            // FullName
+            if (!String.IsNullOrEmpty(envDTEProject.FullName))
```

Comment:
`var path = envDTEProject.FullName;`

Put this in a string first, calling the property twice here should result in double the work, it's hard to say what is happening behind this call, but it is probably different for every project system, and it could be making COM calls to get this

Reply:
nit: lowercase string